src/mirror.c

```diff
@@ -297,7 +297,12 @@ int handle_mirror_if_stale(void)
 
 	/* if the latest version could not be retrieved from the central server we cannot
 	 * check if mirror is stale, this would be very odd since we already check connection
-	 * with the server earlier */
+	 * with the server earlier. The ONLY other possible explanation for this if the signature
+	 * for central_version dont verify
+	 */
+	if (central_version == -SWUPD_SIGNATURE_VERIFICATION_FAILED) {
+		warn("Signature verification for path %s failed\n");
+	}
```

Comment:
I was thinking at such high level in code, it would be good to know what happened with finding the mirror version apart from the signature check failure.

Maybe I can do something like this:

if (central_version == -SWUPD_SIGNATURE_VERIFICATION_FAILED) {
		warn("Signature verification for path %s failed\n", ret_str);
                 warn("Cannot determine upstream version");
                 warn("Unable to determine if the mirror is up to date\n");
                 goto out;
}
else if (central_version < 0){
 / * same as before */
}

Let me know what you think?

Reply:
yeah, I like it better that way, but I would merge the first two warning into one, something like "Cannot determine upstream version since signature verification for path %s failed"

Reply:
make sense..done